From 2c6b15b2a34bdc37e9c420792b9e3adcf07ea933 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <josef.schlehofer@nic.cz>
Date: Tue, 7 Aug 2018 17:45:04 +0200
Subject: [PATCH] debootstrap: update to the version 1.0.106

---
 admin/debootstrap/Makefile                      | 11 +++++-------
 1 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/admin/debootstrap/Makefile b/admin/debootstrap/Makefile
index 47ff1c8e..9a301ba0 100644
--- a/admin/debootstrap/Makefile
+++ b/admin/debootstrap/Makefile
@@ -9,13 +9,13 @@
 include $(TOPDIR)/rules.mk

 PKG_NAME:=debootstrap
-PKG_VERSION:=1.0.81
+PKG_VERSION:=1.0.106
 PKG_RELEASE:=1
-PKG_MAINTAINER=Daniel Golle <daniel@makrotopia.org>
+PKG_MAINTAINER:=Daniel Golle <daniel@makrotopia.org>

 PKG_SOURCE:=$(PKG_NAME)-udeb_$(PKG_VERSION)_all.udeb
 PKG_SOURCE_URL:=http://ftp.debian.org/debian/pool/main/d/debootstrap
-PKG_MD5SUM:=06d2fc8005113da0c74aa4c8ff82d659
+PKG_MD5SUM:=c1adbd46089b1b5f366075213b5339b2
 PKG_LICENSE:=Unique
 PKG_LICENSE_FILES:=debian/copyright

@@ -28,7 +28,7 @@ define Package/debootstrap
   CATEGORY:=Administration
   TITLE:=Bootstrap a basic Debian system
   URL:=http://wiki.debian.org/Debootstrap
-  DEPENDS:= +coreutils +coreutils-chroot +coreutils-sha1sum +ar
+  DEPENDS:= +coreutils +coreutils-chroot +coreutils-sha1sum +ar +xz
 endef

 define Package/debootstrap/description
@@ -49,9 +49,8 @@ define Package/debootstrap/install
 	$(INSTALL_DIR) $(1)/usr/share/debootstrap
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/usr/share/debootstrap/pkgdetails $(1)/usr/share/debootstrap
 	$(INSTALL_DATA) $(PKG_BUILD_DIR)/usr/share/debootstrap/functions $(1)/usr/share/debootstrap
-	$(INSTALL_DATA) $(PKG_BUILD_DIR)/usr/share/debootstrap/devices.tar.gz $(1)/usr/share/debootstrap
 	$(INSTALL_DIR) $(1)/usr/share/debootstrap/scripts
-	$(INSTALL_DATA) $(PKG_BUILD_DIR)/usr/share/debootstrap/scripts/* $(1)/usr/share/debootstrap/scripts
+	$(CP) $(PKG_BUILD_DIR)/usr/share/debootstrap/scripts/* $(1)/usr/share/debootstrap/scripts
 endef

 $(eval $(call BuildPackage,debootstrap))
