diff --git a/libs/giflib/Makefile b/libs/giflib/Makefile
index 28fcd69..75573bb 100644
--- a/libs/giflib/Makefile
+++ b/libs/giflib/Makefile
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=giflib
-PKG_VERSION:=5.1.2
+PKG_VERSION:=5.1.4
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=@SF/giflib
-PKG_MD5SUM:=323a9f11ab56c8a2d1715376410ce376
+PKG_MD5SUM:=2c171ced93c0e83bb09e6ccad8e3ba2b
 
 PKG_MAINTAINER:=Ted Hess <thess@kitschensync.net>
 
