diff --git a/multimedia/gstreamer1/Makefile b/multimedia/gstreamer1/Makefile
index cf11d9d..27ce013 100644
--- a/multimedia/gstreamer1/Makefile
+++ b/multimedia/gstreamer1/Makefile
@@ -29,6 +29,8 @@ PKG_INSTALL:=1
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/nls.mk
 
+TARGET_CFLAGS+= -std=gnu99 -DHAVE_DECL_STRSIGNAL=1
+
 define Package/gstreamer1/Default
   CATEGORY:=Multimedia
   SECTION:=multimedia
