diff --git a/utils/hdparm/Makefile b/utils/hdparm/Makefile
index fabc46439..51f494812 100644
--- a/utils/hdparm/Makefile
+++ b/utils/hdparm/Makefile
@@ -8,13 +8,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=hdparm
-PKG_VERSION:=9.48
+PKG_VERSION:=9.58
 PKG_RELEASE:=1
 PKG_USE_MIPS16:=0
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=@SF/$(PKG_NAME)
-PKG_MD5SUM:=213efdbe7471fad3408198918e164354
+PKG_MD5SUM:=4652c49cf096a64683c05f54b4fa4679
 PKG_MAINTAINER:=Richard Kunze <richard.kunze@web.de>
 PKG_LICENSE:=BSD-Style Open Source License
 
