diff --git a/net/isc-dhcp/Makefile b/net/isc-dhcp/Makefile
index 1de59818..ec16a344 100644
--- a/net/isc-dhcp/Makefile
+++ b/net/isc-dhcp/Makefile
@@ -10,7 +10,7 @@ include $(TOPDIR)/rules.mk
 PKG_NAME:=isc-dhcp
 UPSTREAM_NAME:=dhcp
 PKG_VERSION:=4.3.3
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_LICENSE:=BSD-3-Clause
 PKG_LICENSE_FILES:=LICENSE
@@ -158,6 +158,7 @@ endef
 CONFIGURE_ARGS += \
 	--disable-tracing 		\
 	--enable-paranoia		\
+	--enable-failover		\
 	--disable-dependency-tracking	\
 	--with-randomdev=/dev/urandom \
 	ac_cv_file__dev_random=yes
