diff -Naru a/net/lighttpd/files/lighttpd.conf b/net/lighttpd/files/lighttpd.conf
--- a/net/lighttpd/files/lighttpd.conf	2016-06-27 14:36:55.690329270 +0200
+++ b/net/lighttpd/files/lighttpd.conf	2016-07-04 09:45:18.907902778 +0200
@@ -5,8 +5,8 @@
 server.upload-dirs          = ( "/tmp" )
 server.errorlog             = "/var/log/lighttpd/error.log"
 server.pid-file             = "/var/run/lighttpd.pid"
-server.username             = "http"
-server.groupname            = "www-data"
+#server.username             = "http"
+#server.groupname            = "www-data"
 
 index-file.names            = ( "index.php", "index.html",
                                 "index.htm", "default.htm",
@@ -22,6 +22,9 @@
 #server.errorlog-use-syslog  = "enable"
 #server.network-backend      = "write"
 
+# listen on IPv6
+$SERVER["socket"] == "[::]:80" {  }
+
 ### Use IPv6 if available
 #include_shell "/usr/share/lighttpd/use-ipv6.pl"
 
@@ -29,4 +32,5 @@
 #server.dir-listing          = "enable"
 
 include       "/etc/lighttpd/mime.conf"
+include_shell "cat /etc/lighttpd/modules.d/*.load"
 include_shell "cat /etc/lighttpd/conf.d/*.conf"
diff -Naru a/net/lighttpd/files/lighttpd.init b/net/lighttpd/files/lighttpd.init
--- a/net/lighttpd/files/lighttpd.init	2016-06-27 14:36:55.690329270 +0200
+++ b/net/lighttpd/files/lighttpd.init	2016-07-04 09:45:18.931902954 +0200
@@ -3,14 +3,11 @@
 
 SERVICE_USE_PID=1
 
-START=50
+# Start after the updater resume
+START=80
 
 start() {
-	user_exists http || user_add http
-	[ -d /var/log/lighttpd ] || {
-		mkdir -m 0775 -p /var/log/lighttpd
-		chgrp www-data /var/log/lighttpd
-	}
+	mkdir -m 0755 -p /var/log/lighttpd
 	service_start /usr/sbin/lighttpd -f /etc/lighttpd/lighttpd.conf
 }
 
@@ -18,3 +15,10 @@
 	service_stop /usr/sbin/lighttpd
 }
 
+restart() {
+	/usr/sbin/lighttpd -t -f /etc/lighttpd/lighttpd.conf || exit 1
+	ls /usr/lib/lighttpd/mod_alias.so || exit 1
+	ls /usr/lib/lighttpd/mod_cgi.so || exit 1
+	stop
+	start
+}
diff -Naru a/net/lighttpd/Makefile b/net/lighttpd/Makefile
--- a/net/lighttpd/Makefile	2016-07-04 09:47:54.641047847 +0200
+++ b/net/lighttpd/Makefile	2016-07-04 09:45:18.907902778 +0200
@@ -38,6 +38,18 @@
   TITLE:=A flexible and lightweight web server
 endef
 
+define Package/lighttpd/postinst
+#!/bin/sh
+[ -n "$$IPKG_INSTROOT" ] || {
+if [ -e /etc/rc.d/S86lighttpd ] ; then
+	/etc/init.d/lighttpd disable
+	/etc/init.d/lighttpd enable
+fi
+# During initial installation, there are still some modules missing. It'll be started by something else later.
+/etc/init.d/lighttpd restart || ( sleep 5 ; /etc/init.d/lighttpd restart ) || true
+}
+endef
+
 define Package/lighttpd/config
 config LIGHTTPD_SSL
 	bool "SSL support"
@@ -113,6 +125,7 @@
 	$(INSTALL_DIR) $(1)/etc/lighttpd
 	$(INSTALL_DATA) ./files/lighttpd.conf $(1)/etc/lighttpd/
 	$(INSTALL_DATA) $(PKG_BUILD_DIR)/doc/config/conf.d/mime.conf $(1)/etc/lighttpd/
+	$(INSTALL_DIR) $(1)/etc/lighttpd/modules.d
 	$(INSTALL_DIR) $(1)/etc/lighttpd/conf.d
 	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) ./files/lighttpd.init $(1)/etc/init.d/lighttpd
@@ -137,12 +150,8 @@
   define Package/lighttpd-mod-$(1)/install
 	$(INSTALL_DIR) $$(1)/usr/lib/lighttpd
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/lighttpd/mod_$(1).so $$(1)/usr/lib/lighttpd
-	$(INSTALL_DIR) $$(1)/etc/lighttpd/conf.d
-	if [ -f $(PKG_BUILD_DIR)/doc/config/conf.d/$(1).conf ]; then \
-		$(CP) $(PKG_BUILD_DIR)/doc/config/conf.d/$(1).conf $$(1)/etc/lighttpd/conf.d/$(4)-$(1).conf ; \
-	else \
-		echo 'server.modules += ( "mod_$(1)" )' > $$(1)/etc/lighttpd/conf.d/$(4)-$(1).conf ; \
-	fi
+	$(INSTALL_DIR) $$(1)/etc/lighttpd/modules.d/
+	echo 'server.modules += ( "mod_$(1)" )' > $$(1)/etc/lighttpd/modules.d/$(4)-$(1).load
   endef
 
   $$(eval $$(call BuildPackage,lighttpd-mod-$(1)))
