diff --git a/libs/openldap/Makefile b/libs/openldap/Makefile
index f9a8347..b7412e6 100644
--- a/libs/openldap/Makefile
+++ b/libs/openldap/Makefile
@@ -37,7 +37,7 @@ define Package/libopenldap
   $(call Package/openldap/Default)
   SECTION:=libs
   CATEGORY:=Libraries
-  DEPENDS:=+libopenssl +libsasl2 +libpthread
+  DEPENDS:=+libopenssl +libsasl2 +libpthread +libicu
   TITLE+= (libraries)
 endef
 
@@ -67,7 +67,7 @@ define Package/openldap-server
   $(call Package/openldap/Default)
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:=+libopenldap +libuuid
+  DEPENDS:=+libopenldap +libuuid +libicu
   TITLE+= (server)
 endef
 
