diff --git a/lang/php5/Makefile b/lang/php5/Makefile
index ac1aa14..ed238c1 100644
--- a/lang/php5/Makefile
+++ b/lang/php5/Makefile
@@ -8,7 +8,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=php
-PKG_VERSION:=5.6.21
+PKG_VERSION:=5.6.29
 PKG_RELEASE:=1
 
 PKG_MAINTAINER:=W. Michael Petullo <mike@flyn.org>, Michael Heimpold <mhei@heimpold.de>
@@ -18,7 +18,7 @@ PKG_LICENSE_FILES:=LICENSE
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=http://www.php.net/distributions/
-PKG_MD5SUM:=177c69d47024541739c439c9d9eb6ba5
+PKG_MD5SUM:=190bf5b52d1fc68d5500a8cdc7e33164
 
 PKG_FIXUP:=libtool autoreconf
 PKG_BUILD_PARALLEL:=1
