diff --git a/libs/protobuf/Makefile b/libs/protobuf/Makefile
index 1e70a12..72133be 100644
--- a/libs/protobuf/Makefile
+++ b/libs/protobuf/Makefile
@@ -41,7 +41,7 @@ define Package/protobuf/description
  of its internal RPC protocols and file formats.
 endef
 
-CONFIGURE_ARGS += --with-protoc=$(STAGING_DIR)/host/bin/protoc
+CONFIGURE_ARGS += --with-protoc=$(STAGING_DIR_HOST)/bin/protoc
 
 define Build/InstallDev
 	$(INSTALL_DIR) \
