diff --git a/lang/python-cryptography/Makefile b/lang/python-cryptography/Makefile
index 79875cf1..0d108272 100644
--- a/lang/python-cryptography/Makefile
+++ b/lang/python-cryptography/Makefile
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=cryptography
-PKG_VERSION:=1.3.1
+PKG_VERSION:=2.3
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=https://pypi.python.org/packages/source/c/cryptography
-PKG_MD5SUM:=bc8148d2ff2d80fef8ef2d2e856b3a7f
+PKG_SOURCE_URL:=https://github.com/pyca/cryptography/archive/
+PKG_MD5SUM:=351378eb01b66d06815206fff9b4ec0e
 
 PKG_BUILD_DEPENDS:=python-cffi/host
 
@@ -24,6 +24,16 @@ PKG_MAINTAINER:=Jeffery To <jeffery.to@gmail.com>
 include $(INCLUDE_DIR)/package.mk
 $(call include_mk, python-package.mk)
 
+define Download/default
+  FILE:=$(PKG_SOURCE)
+  URL:=$(PKG_SOURCE_URL)
+  URL_FILE:=$(PKG_VERSION).tar.gz
+  SUBDIR:=$(PKG_SOURCE_SUBDIR)
+  PROTO:=$(PKG_SOURCE_PROTO)
+  VERSION:=$(PKG_SOURCE_VERSION)
+  MD5SUM:=$(PKG_MD5SUM)
+endef
+
 define Package/python-cryptography
 	SECTION:=lang
 	CATEGORY:=Languages
