diff --git a/net/rsyslog/Makefile b/net/rsyslog/Makefile
index 6add57c..465e7df 100644
--- a/net/rsyslog/Makefile
+++ b/net/rsyslog/Makefile
@@ -29,7 +29,7 @@ define Package/rsyslog
   CATEGORY:=Network
   TITLE:=Enhanced system logging and kernel message trapping daemons
   URL:=http://www.rsyslog.com/
-  DEPENDS:=+libestr +libjson-c +libuuid +zlib
+  DEPENDS:=+libestr +libjson-c +libuuid +zlib +librt +libpthread
 endef
 
 define Package/rsyslog/conffiles
