diff --git a/utils/smartmontools/Makefile b/utils/smartmontools/Makefile
index 88d684020..5403a55ac 100644
--- a/utils/smartmontools/Makefile
+++ b/utils/smartmontools/Makefile
@@ -9,12 +9,12 @@ include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/uclibc++.mk
 
 PKG_NAME:=smartmontools
-PKG_VERSION:=6.5
-PKG_RELEASE:=1
+PKG_VERSION:=7.0
+PKG_RELEASE:=0
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=@SF/smartmontools
-PKG_MD5SUM:=093aeec3f8f39fa9a37593c4012d3156
+PKG_MD5SUM:=b2a80e4789af23d67dfe1e88a997abbf
 PKG_MAINTAINER:=Maxim Storchak <m.storchak@gmail.com>
 PKG_LICENSE:=GPL-2.0+
 PKG_LICENSE_FILES:=COPYING
