diff --git a/net/usbip/Makefile b/net/usbip/Makefile
index 3148514f..64d926e8 100644
--- a/net/usbip/Makefile
+++ b/net/usbip/Makefile
@@ -30,7 +30,7 @@ define prepare_source_directory
 endef
 Hooks/Prepare/Pre += prepare_source_directory
 
-PKG_BUILD_DEPENDS:=eudev
+PKG_BUILD_DEPENDS:=udev
 PKG_FIXUP:=autoreconf
 PKG_INSTALL:=1
 PKG_FLAGS:=nonshared
@@ -53,7 +53,7 @@ endef
 define Package/usbip
   $(call Package/usbip/Default)
   TITLE+= (common)
-  DEPENDS+= +libwrap +kmod-usbip +eudev
+  DEPENDS+= +libwrap +kmod-usbip +udev
 endef
 
 define Package/usbip-client
