--- a/init/do_mounts.c
+++ b/init/do_mounts.c
@@ -307,10 +307,13 @@
 	return 1;
 }
 
-static char * __initdata root_fs_names;
+static char * __initdata root_fs_names = "squashfs,jffs2";
 static int __init fs_names_setup(char *str)
 {
 	root_fs_names = str;
+
+        printk("DEEEEBUG: fs_names_setup: %s \n", root_fs_names);
+
 	return 1;
 }
 
@@ -350,11 +353,14 @@
 		}
 	}
 	*s = '\0';
+
+        printk("get_fs_names: %s \n", s);
 }
 
 static int __init do_mount_root(char *name, char *fs, int flags, void *data)
 {
 	struct super_block *s;
+	printk("do_mount_root: %s || %s\n", name, fs);
 	int err = sys_mount(name, "/root", fs, flags, data);
 	if (err)
 		return err;
