--- a/libopkg/opkg_download.c
+++ b/libopkg/opkg_download.c
@@ -90,27 +90,19 @@ opkg_download(const char *src, const cha
 		const char *argv[8];
 		int i = 0;
 
-		argv[i++] = "wget";
-		argv[i++] = "-q";
-		if (conf->http_proxy || conf->ftp_proxy) {
-			argv[i++] = "-Y";
-			argv[i++] = "on";
-		}
-		argv[i++] = "-O";
+		i = 0;
+
+		argv[i++] = "curl";
+		argv[i++] = "--compress";
+		argv[i++] = "-f";
+		argv[i++] = "-o";
 		argv[i++] = tmp_file_location;
 		argv[i++] = src;
 		argv[i++] = NULL;
 		res = xsystem(argv);
 
 		if (res) {
-			opkg_msg(ERROR,
-				 "Failed to download %s, wget returned %d.\n",
-				 src, res);
-			if (res == 4)
-				opkg_msg(ERROR,
-					 "Check your network settings and connectivity.\n\n");
-			free(tmp_file_location);
-			return -1;
+			opkg_msg(ERROR, "Failed to download %s, curl returned %d.\n", src, res);
 		}
 	}
 
