--- a/libopkg/opkg_cmd.c
+++ b/libopkg/opkg_cmd.c
@@ -492,6 +492,16 @@ static int opkg_upgrade_cmd(int argc, ch
 	pkg_t *pkg;
 	int err = 0;
 
+	char answer[5];
+	fprintf(stderr, "The opkg upgrade is *DANGEROUS* on TurrisOS releases.\n"
+	                "It *WILL* break your system.\n"
+	                "Do you really want to continue? Write upper-case yes if do.\n");
+
+	fgets(answer, 5, stdin);
+
+	if (strcmp("YES\n", answer))
+		exit(1);
+
 	signal(SIGINT, sigint_handler);
 
 	if (argc) {
