Index: opkg-9c97d5ecd795709c8584e972bfdf3aee3a5b846d/libopkg/pkg.c
===================================================================
--- opkg-9c97d5ecd795709c8584e972bfdf3aee3a5b846d.orig/libopkg/pkg.c
+++ opkg-9c97d5ecd795709c8584e972bfdf3aee3a5b846d/libopkg/pkg.c
@@ -734,6 +734,7 @@ pkg_formatted_field(FILE *fp, pkg_t *pkg
 	       if (pkg->provides_count > 1) {
                   fprintf(fp, "Provides:");
 		  for(i = 1; i < pkg->provides_count; i++) {
+                      if(pkg->provides[i] != NULL)
                       fprintf(fp, "%s %s", i == 1 ? "" : ",",
 				      pkg->provides[i]->name);
                   }
Index: opkg-9c97d5ecd795709c8584e972bfdf3aee3a5b846d/libopkg/opkg_cmd.c
===================================================================
--- opkg-9c97d5ecd795709c8584e972bfdf3aee3a5b846d.orig/libopkg/opkg_cmd.c
+++ opkg-9c97d5ecd795709c8584e972bfdf3aee3a5b846d/libopkg/opkg_cmd.c
@@ -942,7 +942,8 @@ pkg_mark_provides(pkg_t *pkg)
      int i;
      pkg->parent->state_flag |= SF_MARKED;
      for (i = 0; i < provides_count; i++) {
-	  provides[i]->state_flag |= SF_MARKED;
+          if(provides[i] != NULL)
+	      provides[i]->state_flag |= SF_MARKED;
      }
      return 0;
 }
