--- a/libopkg/pkg.c
+++ b/libopkg/pkg.c
@@ -734,6 +734,7 @@ pkg_formatted_field(FILE *fp, pkg_t *pkg
 	       if (pkg->provides_count > 1) {
                   fprintf(fp, "Provides:");
 		  for(i = 1; i < pkg->provides_count; i++) {
+                      if(pkg->provides[i] != NULL)
                       fprintf(fp, "%s %s", i == 1 ? "" : ",",
 				      pkg->provides[i]->name);
                   }
