--- a/Configure
+++ b/Configure
@@ -1303,7 +1303,9 @@ else               { $no_user_defines=1;
 
 unless ($disabled{afalgeng}) {
     $config{afalgeng}="";
-    if ($target =~ m/^linux/) {
+    if ($target =~ m/openwrt$/) {
+        push @{$config{engdirs}}, "afalg";
+    } elsif ($target =~ m/^linux/) {
         my $minver = 4*10000 + 1*100 + 0;
         if ($config{cross_compile_prefix} eq "") {
             my $verstr = `uname -r`;
