--- a/net/wireless/reg.c
+++ b/net/wireless/reg.c
@@ -90,6 +90,10 @@ enum reg_request_treatment {
 	REG_REQ_ALREADY_SET,
 };
 
+static char* freg = "**";
+module_param(freg, charp, 0644);
+MODULE_PARM_DESC(freg,
+		 "Ignores users requests to set any other regulatory domain");
 static struct regulatory_request core_request_world = {
 	.initiator = NL80211_REGDOM_SET_BY_CORE,
 	.alpha2[0] = '0',
@@ -2370,8 +2374,13 @@ int regulatory_hint_user(const char *alp
 		return -ENOMEM;
 
 	request->wiphy_idx = WIPHY_IDX_INVALID;
-	request->alpha2[0] = alpha2[0];
-	request->alpha2[1] = alpha2[1];
+	if(freg[0] != '*') { printk("REG domain set to %c%c instead of %c%c\n", freg[0], freg[1], alpha2[0], alpha2[1]);
+		request->alpha2[0] = freg[0];
+		request->alpha2[1] = freg[1];
+	} else { printk("REG domain set to %c%c (freg = %c%c)\n", alpha2[0], alpha2[1], freg[0], freg[1]);
+		request->alpha2[0] = alpha2[0];
+		request->alpha2[1] = alpha2[1];
+	}
 	request->initiator = NL80211_REGDOM_SET_BY_USER;
 	request->user_reg_hint_type = user_reg_hint_type;
 
