--- a/tcpdump.c
+++ b/tcpdump.c
@@ -257,27 +257,6 @@ __declspec(dllimport)
 extern
 #endif /* _WIN32 */
 void pcap_set_parser_debug(int);
-#elif defined(HAVE_PCAP_DEBUG) || defined(HAVE_YYDEBUG)
-/*
- * We don't have pcap_set_parser_debug() in libpcap, but we do have
- * pcap_debug or yydebug.  Make a local version of pcap_set_parser_debug()
- * to set the flag, and define HAVE_PCAP_SET_PARSER_DEBUG.
- */
-static void
-pcap_set_parser_debug(int value)
-{
-#ifdef HAVE_PCAP_DEBUG
-	extern int pcap_debug;
-
-	pcap_debug = value;
-#else /* HAVE_PCAP_DEBUG */
-	extern int yydebug;
-
-	yydebug = value;
-#endif /* HAVE_PCAP_DEBUG */
-}
-
-#define HAVE_PCAP_SET_PARSER_DEBUG
 #endif
 
 #if defined(HAVE_PCAP_SET_OPTIMIZER_DEBUG)
