Index: uci-2015-04-09.1/file.c
===================================================================
--- uci-2015-04-09.1.orig/file.c
+++ uci-2015-04-09.1/file.c
@@ -721,6 +721,7 @@ static void uci_file_commit(struct uci_c
 {
 	struct uci_package *p = *package;
 	FILE *f1, *f2 = NULL;
+	int syncfd = 0;
 	char *name = NULL;
 	char *path = NULL;
 	char *filename = NULL;
@@ -796,13 +797,16 @@ static void uci_file_commit(struct uci_c
 done:
 	free(name);
 	free(path);
+	syncfs(fileno(f1));
 	uci_close_stream(f1);
 	if (do_rename && rename(filename, p->path)) {
+		syncfd = open(filename, O_RDONLY | O_NONBLOCK);
+		if (syncfd > 0)
+			syncfs(syncfd);
 		unlink(filename);
 		UCI_THROW(ctx, UCI_ERR_IO);
 	}
 	free(filename);
-	sync();
 	if (ctx->err)
 		UCI_THROW(ctx, ctx->err);
 }
