--- a/drivers/net/ethernet/marvell/mvneta.c
+++ b/drivers/net/ethernet/marvell/mvneta.c
@@ -3080,10 +3080,12 @@ static int mvneta_prepare_phy(struct pla
 			return -ENODEV;
 		}
 
-		err = of_phy_register_fixed_link(mode_dn);
-		if (err < 0) {
-			dev_err(&pdev->dev, "cannot register fixed PHY\n");
-			return -ENODEV;
+		if (!of_phy_find_device(mode_dn)) {
+			err = of_phy_register_fixed_link(mode_dn);
+			if (err < 0) {
+				dev_err(&pdev->dev, "cannot register fixed PHY\n");
+				return -ENODEV;
+			}
 		}
 
 		/* In the case of a fixed PHY, the DT node associated
