Index: linux-4.4.44/drivers/w1/masters/w1-gpio.c
===================================================================
--- linux-4.4.44.orig/drivers/w1/masters/w1-gpio.c
+++ linux-4.4.44/drivers/w1/masters/w1-gpio.c
@@ -88,24 +88,26 @@ static int w1_gpio_probe_dt(struct platf
 	if (of_get_property(np, "linux,open-drain", NULL))
 		pdata->is_open_drain = 1;
 
-	gpio = of_get_gpio(np, 0);
-	if (gpio < 0) {
-		if (gpio != -EPROBE_DEFER)
-			dev_err(&pdev->dev,
+	if (pdev->dev.of_node) {
+		gpio = of_get_gpio(np, 0);
+		if (gpio < 0) {
+			if (gpio != -EPROBE_DEFER)
+				dev_err(&pdev->dev,
 					"Failed to parse gpio property for data pin (%d)\n",
 					gpio);
 
-		return gpio;
-	}
-	pdata->pin = gpio;
+			return gpio;
+		}
+		pdata->pin = gpio;
 
-	gpio = of_get_gpio(np, 1);
-	if (gpio == -EPROBE_DEFER)
-		return gpio;
-	/* ignore other errors as the pullup gpio is optional */
-	pdata->ext_pullup_enable_pin = gpio;
+		gpio = of_get_gpio(np, 1);
+		if (gpio == -EPROBE_DEFER)
+			return gpio;
+		/* ignore other errors as the pullup gpio is optional */
+		pdata->ext_pullup_enable_pin = gpio;
 
-	pdev->dev.platform_data = pdata;
+		pdev->dev.platform_data = pdata;
+	}
 
 	return 0;
 }
