diff --git a/lang/php7/files/php.ini b/lang/php7/files/php.ini
index c88ab0a4..d08815a0 100644
--- a/lang/php7/files/php.ini
+++ b/lang/php7/files/php.ini
@@ -42,7 +42,7 @@ expose_php = On
 max_execution_time = 30	; Maximum execution time of each script, in seconds.
 max_input_time = 60	; Maximum amount of time each script may spend parsing request data.
 ;max_input_nesting_level = 64
-memory_limit = 8M	; Maximum amount of memory a script may consume.
+memory_limit = 128M	; Maximum amount of memory a script may consume.
 
 ; Error handling and logging
 
@@ -123,7 +123,7 @@ default_mimetype = "text/html"
 
 ; UNIX: "/path1:/path2"
 ;include_path = ".:/php/includes"
-doc_root = "/www"
+doc_root = "/srv/www"
 user_dir =
 extension_dir = "/usr/lib/php"
 enable_dl = On
@@ -139,7 +139,7 @@ cgi.fix_pathinfo=1
 
 file_uploads = On
 upload_tmp_dir = "/tmp"
-upload_max_filesize = 2M
+upload_max_filesize = 256M
 max_file_uploads = 20
 
 ; Fopen wrappers
diff --git a/lang/php7/files/opcache.ini b/lang/php7/files/opcache.ini
new file mode 100644
index 00000000..031b26fc
--- /dev/null
+++ b/lang/php7/files/opcache.ini
@@ -0,0 +1,7 @@
+opcache.enable=1
+opcache.enable_cli=1
+opcache.interned_strings_buffer=8
+opcache.max_accelerated_files=10000
+opcache.memory_consumption=128
+opcache.save_comments=1
+opcache.revalidate_freq=1
diff --git a/lang/php7/Makefile b/lang/php7/Makefile
index f7bdd66e..922f5ccc 100644
--- a/lang/php7/Makefile
+++ b/lang/php7/Makefile
@@ -537,6 +540,7 @@ define BuildModule
 	$(INSTALL_DIR) $$(1)/usr/lib/php
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/modules/$(subst -,_,$(1)).so $$(1)/usr/lib/php/
 	$(INSTALL_DIR) $$(1)/etc/php7
+	[ \! -f ./files/$(subst -,_,$(1)).ini ] || cp ./files/$(subst -,_,$(1)).ini $$(1)/etc/php7/99_$(subst -,_,$(1)).ini
       ifeq ($(5),zend)
 	echo "zend_extension=/usr/lib/php/$(subst -,_,$(1)).so" > $$(1)/etc/php7/$(if $(4),$(4),20)_$(subst -,_,$(1)).ini
       else
